File: apps/microblog/extension/fable_output/packages/hedge/src/Hedge/Tenant.js
import { Record } from "../../../../fable_modules/fable-library-js.4.29.0/Types.js";
import { record_type, class_type, string_type } from "../../../../fable_modules/fable-library-js.4.29.0/Reflection.js";
import { FSharpSet__Contains, ofArray } from "../../../../fable_modules/fable-library-js.4.29.0/Set.js";
import { map } from "../../../../fable_modules/fable-library-js.4.29.0/Array.js";
import { comparePrimitives } from "../../../../fable_modules/fable-library-js.4.29.0/Util.js";

export class TenantConfig extends Record {
    constructor(Slug, Title, Logo, Features) {
        super();
        this.Slug = Slug;
        this.Title = Title;
        this.Logo = Logo;
        this.Features = Features;
    }
}

export function TenantConfig_$reflection() {
    return record_type("Hedge.Tenant.TenantConfig", [], TenantConfig, () => [["Slug", string_type], ["Title", string_type], ["Logo", string_type], ["Features", class_type("Microsoft.FSharp.Collections.FSharpSet`1", [string_type])]]);
}

export const config = new TenantConfig((window.SITE_SLUG || ''), (window.SITE_TITLE || ''), (window.SITE_LOGO || ''), ofArray((() => {
    const array_1 = map((s) => s.trim(), ((window.SITE_FEATURES || '')).split(","));
    return array_1.filter((s_1) => (s_1 !== ""));
})(), {
    Compare: comparePrimitives,
}));

/**
 * Is a per-tenant feature enabled? (from window.SITE_FEATURES, a comma list)
 */
export function hasFeature(feature) {
    return FSharpSet__Contains(config.Features, feature);
}

File: packages/hedge/src/Hedge/Tenant.fs
module Hedge.Tenant

// Per-tenant config, delivered to the client at build time: the app's
// vite siteConfig plugin injects window.SITE_* (and stamps body.tenant-<slug>),
// and this reads them into one typed accessor. Per-tenant *behaviour* gates on a
// feature flag (Tenant.hasFeature "x") rather than hardcoded tenant-slug checks.

open Fable.Core

type TenantConfig = {
    Slug: string
    Title: string
    Logo: string
    Features: Set<string>
    /// Optional external info/companion page (e.g. a campaign page on Pages). When
    /// set, the client shows a prominent link to it. Empty = no link.
    InfoUrl: string
    InfoLabel: string
}

[<Emit("(window.SITE_SLUG || '')")>]
let private slug : string = jsNative
[<Emit("(window.SITE_TITLE || '')")>]
let private title : string = jsNative
[<Emit("(window.SITE_LOGO || '')")>]
let private logo : string = jsNative
[<Emit("(window.SITE_FEATURES || '')")>]
let private featuresRaw : string = jsNative
[<Emit("(window.SITE_INFO_URL || '')")>]
let private infoUrl : string = jsNative
[<Emit("(window.SITE_INFO_LABEL || '')")>]
let private infoLabel : string = jsNative

let config : TenantConfig =
    { Slug = slug
      Title = title
      Logo = logo
      InfoUrl = infoUrl
      InfoLabel = infoLabel
      Features =
        featuresRaw.Split(',')
        |> Array.map (fun s -> s.Trim())
        |> Array.filter (fun s -> s <> "")
        |> Set.ofArray }

/// Is a per-tenant feature enabled? (from window.SITE_FEATURES, a comma list)
let hasFeature (feature: string) : bool = config.Features.Contains feature
